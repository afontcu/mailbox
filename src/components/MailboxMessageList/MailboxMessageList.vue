<template>
  <aside class="MailboxMessageList">
    <p class="MailboxMessageList__title">
      {{ unreadMessagesCount }} unread 
      {{ pluralize('message') }}
    </p>
    <mailbox-message-list-item v-for="message in messageList" 
                               :key="message.uid" 
                               v-bind="message"
    />
  </aside>
</template>

<script>
import MailboxMessageListItem from '@/components/MailboxMessageListItem'
import { mapState, mapGetters } from 'vuex'
import { maybePluralize } from '@/common/string'

export default {
  name: 'MailboxMessageList',
  components: {
    MailboxMessageListItem,
  },
  props: {
    messageList: { type: Array, required: true },
  },
  computed: {
    ...mapState(['readMessages']),
    ...mapGetters(['messages', 'unreadMessagesCount']),
  },
  methods: {
    pluralize(str) {
      return maybePluralize(this.unreadMessagesCount, str)
    },
  },
}
</script>

<style scoped>
.MailboxMessageList__title {
  font-style: italic;
  margin-bottom: var(--global-spacing);
}
</style>
