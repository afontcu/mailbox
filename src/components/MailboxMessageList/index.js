import MailboxMessageList from './MailboxMessageList'

export default MailboxMessageList
