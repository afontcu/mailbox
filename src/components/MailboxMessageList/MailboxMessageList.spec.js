import MailboxMessageList from './index.js'
import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

const propsData = {
  messageList: [],
}

describe('MailboxMessageList.vue', () => {
  let store, getters, state

  beforeEach(() => {
    state = {
      messages: [],
      readMessages: [],
    }
    getters = {
      messages: () => [],
      unreadMessagesCount: () => 3,
    }
    store = new Vuex.Store({
      getters,
      state,
    })
  })

  it('is a component', () => {
    const wrapper = mount(MailboxMessageList, { propsData, store, localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(MailboxMessageList, { propsData, store, localVue })
    expect(wrapper.classes()).toContain('MailboxMessageList')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(MailboxMessageList, { propsData, store, localVue })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
