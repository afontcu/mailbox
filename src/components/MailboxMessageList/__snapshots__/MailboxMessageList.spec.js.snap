// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailboxMessageList.vue renders a snapshot: default 1`] = `
<aside class="MailboxMessageList">
  <p class="MailboxMessageList__title">
    3 unread
    messages
  </p>
</aside>
`;
